examples/nidaqmx/analog-output.py

```diff
@@ -44,9 +44,8 @@
 def RaiseIfError(response):
     if response.status != 0:
         response = client.GetErrorString(nidaqmx_types.GetErrorStringRequest(
```

Comment:
@harsha-bhushan : we figured out that the `ivi-dance` works for a lot of DAQ methods. This removed the need to pass in a size and also avoids the bug with passed-in array sizes where you have to strip trailing null characters.

What do you think about adding this to the DAQ server?

If we *don't* add this. One problem is that users would implement the ivi-dance themselves by first calling with a zero size and then calling again with the returned size. So, if we tried to add this feature later (by making size optional/deprecated and doing the ivi-dance if it's zero) then we would break those existing usages. Of course, it would be easy for them to catch/fix.

It should be easy to automatically detect *most* of the cases where this would work. We'd have to do some double-checking to make sure that the messages we mark actually do support the convention. And there are cases where multiple arrays are sized together (or separately) that are a little trickier.

A variation would be to leave the size parameter on the proto request. Then we'd use that size if it's passed in (non-zero), but auto-detect the size if it wasn't. The biggest advantage of that is that the method would still be usable if we were wrong about it supporting the ivi-dance.

Reply:
I agree. This would greatly improve the usability. 
We should also run this by/keep informed someone from the team that owns/maintains DAQmx API especially for using ivi-dance on other methods.